pkg/solana/client/multinode/transaction_sender.go

```diff
@@ -145,89 +152,92 @@ func (txSender *TransactionSender[TX, CHAIN_ID, RPC]) SendTransaction(ctx contex
 	}()
 
 	if err != nil {
-		return Retryable, err
+		return txSender.newResult(err)
 	}
 
 	txSender.wg.Add(1)
-	go txSender.reportSendTxAnomalies(tx, txResultsToReport)
+	reportWg.Add(1)
+	go func() {
+		txSender.reportSendTxAnomalies(tx, txResultsToReport)
+		reportWg.Done()
```

Comment:
It's important to defer these so that they still execute on panic:
```suggestion
		defer reportWg.Done()
		txSender.reportSendTxAnomalies(tx, txResultsToReport)
```